### PR TITLE
make spectrogram augmenter optional

### PR DESCRIPTION
**Pull request by zhiyun**

### CODE REVIEW

**Comment by zhiyun** [comment]
The same as https://github.com/apple/axlearn/pull/257, after CI team has fixed my account issue.

**Review by yqwangustc — APPROVED** [review]
Thank you, @zhiyun.  Looks good to me.

File: axlearn/audio/asr_encoder_test.py [hunk 1]
```diff
@@ -243,6 +248,8 @@ def test_asr_encoder(self, is_training: bool):
         self.assertEqual(output_paddings.shape, (batch_size, output_shape[1]))
         self.assertTrue(jnp.all(output_paddings[:2] == output_paddings[2:]))
 
-        # If is_training, outputs should always be different due to augmentation.
+        # If is_training and use_augmenter, outputs should always be different due to augmentation.
         # Otherwise, outputs should be the same despite differences in padding.
-        self.assertEqual(not is_training, jnp.allclose(outputs[:2], outputs[2:]))
+        self.assertEqual(
+            not (is_training and use_augmenter), jnp.allclose(outputs[:2], outputs[2:])
```
Comment (yqwangustc) [inline axlearn/audio/asr_encoder_test.py:254]: nit: I am a little confused about why `outputs[:2]` should be the same as `outputs[2:]`, could you add a comment to explain why they are the same in this case ?

**Review by yqwangustc — APPROVED** [review]

File: axlearn/audio/asr_encoder_test.py (hunk 1 above)
Comment (zhiyun) [reply axlearn/audio/asr_encoder_test.py:254]: Done.

**Comment by yqwangustc** [comment]
approve!

**Review by yqwangustc — APPROVED** [review]

**Pull request merged**